tokio/src/sync/semaphore.rs

```diff
@@ -102,6 +76,54 @@ use std::sync::Arc;
 /// }
 /// ```
 ///
+/// Limit the number of incoming requests being handled at the same time.
+///
+/// Similar to limiting the number of simultaneously opened files, network handles
+/// are a limited resource. Allowing an unbounded amount of requests to be processed
+/// could result in a denial-of-service, among many other issues.
+///
+/// However, in contrast to the file example, this example uses an `Arc<Semaphore>`.
+/// Before we spawn a new task to process a request, we must acquire a permit from
+/// this semaphore, in order to limit usage of system resources (like file descriptors,
+/// memory usage, CPU time, etc.). Once acquired, a new task is spawned; and once
+/// finished, the permit is dropped inside of the task to allow others to spawn.
+/// Permits must be acquired via [`Semaphore::acquire_owned`], in order to be
+/// movable across the task boundary. Contrastingly, if the semaphore were a global,
```

Comment:
You can almost always replace "in order to" with "to".
```suggestion
/// Permits must be acquired via [`Semaphore::acquire_owned`] to be
/// movable across the task boundary. Contrastingly, if the semaphore were a global,
```

Reply:
Yes, I need to improve on my technical writing.